File: app/src/main/java/com/uralsiberianworks/neuralpushkin/api/PushkinApi.java
package com.uralsiberianworks.neuralpushkin.api;

import com.uralsiberianworks.neuralpushkin.database.Message;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface PushkinApi {

    @GET("http://46.17.97.44:5000/push/")
    Call<PushkinResponse> getPushkinExcerption(@Query("text") String text, @Query("temp") double temp, @Query("length") int length);

    @POST("http://46.17.97.44:5000/waifu/")
    Call<BotResponse> getBotResponse(@Body BotRequestBody botRequestBody);
}

File: app/src/main/java/com/uralsiberianworks/neuralpushkin/api/PushkinApi.kt
package com.uralsiberianworks.neuralpushkin.api

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface PushkinApi {
    @GET("http://46.17.97.44:5000/push/")
    fun getPushkinExcerption(@Query("text") text: String?, @Query("temp") temp: Double, @Query("length") length: Int): Call<PushkinResponse?>?

    @GET("http://46.17.97.44:5001/stih/")
    fun getPushkinPoem(@Query("name") name: String?, @Query("temp") temp: Double, @Query("length") length: Int): Call<PushkinResponse?>?
}